lib/publiccloud/aws.pm

```diff
@@ -66,6 +67,10 @@ sub init {
         save_tmp_file(CREDENTIALS_FILE, $credentials_file);
         assert_script_run('curl -O ' . autoinst_url . "/files/" . CREDENTIALS_FILE);
     }
+
+    $self->{aws_account_id} = script_output("aws sts get-caller-identity | jq -r '.Account'");
+    die("Cannot get the UserID")                                        unless ($self->{aws_account_id});
+    die("The UserID doesn't have the correct format: $self->{user_id}") unless length($self->{aws_account_id}) == 12;
```

Comment:
I think it was 12 digits but this is nitty!
```suggestion
    die("The UserID doesn't have the correct format: $self->{user_id}") unless $self->{aws_account_id} =~ /^\d{12}$;
```

Reply:
done